pkg/asset/installconfig/aws/validation.go

```diff
@@ -461,27 +532,187 @@ func validateMachineNetworksContainIP(fldPath *field.Path, networks []types.Mach
+		// We validate edge subnets in subsequent validations.
+		if _, ok := subnetDataGroups.Edge[ingressSubnet.ID]; ok {
+			continue
+		}
+
+		if ingressSubnet.Public != (publish == types.ExternalPublishingStrategy) {
+			subnetType := "private"
+			if ingressSubnet.Public {
+				subnetType = "public"
+			}
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(ingressSubnet.Idx), ingressSubnet.ID,
+				fmt.Sprintf("subnet %s has role %s and is %s, which is not allowed when publish is set to %s", ingressSubnet.ID, awstypes.IngressControllerLBSubnetRole, subnetType, publish)))
+		}
+	}
+
+	// IngressControllerLB subnets must be in different AZs as required by AWS CCM.
+	ingressZones := make(map[string]string)
+	for _, subnetData := range subnetsWithRole[awstypes.IngressControllerLBSubnetRole] {
+		if conflictingSubnet, ok := ingressZones[subnetData.Zone.Name]; ok {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(subnetData.Idx), subnetData.ID,
+				fmt.Sprintf("subnets %s and %s have role %s and are both in zone %s", conflictingSubnet, subnetData.ID, awstypes.IngressControllerLBSubnetRole, subnetData.Zone.Name)))
+		} else {
+			ingressZones[subnetData.Zone.Name] = subnetData.ID
+		}
+	}
+
+	// AZs of LB subnets match AZs of ClusterNode subnets.
+	lbRoles := []awstypes.SubnetRoleType{
+		awstypes.ControlPlaneInternalLBSubnetRole,
+		awstypes.IngressControllerLBSubnetRole,
+	}
+	if publish == types.ExternalPublishingStrategy {
+		lbRoles = append(lbRoles, awstypes.ControlPlaneExternalLBSubnetRole)
+	}
+	for _, role := range lbRoles {
+		allErrs = append(allErrs, validateLBSubnetAZMatchClusterNodeAZ(fldPath, subnetDataGroups, role, subnetsWithRole[role], subnetsWithRole[awstypes.ClusterNodeSubnetRole])...)
+	}
+
+	// EdgeNode subnets must be subnets in Local or Wavelength Zones.
+	for _, edgeSubnet := range subnetsWithRole[awstypes.EdgeNodeSubnetRole] {
+		if _, ok := subnetDataGroups.Edge[edgeSubnet.ID]; !ok {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(edgeSubnet.Idx), edgeSubnet.ID,
+				fmt.Sprintf("subnet %s has role %s, but is not in a Local or WaveLength Zone", edgeSubnet.ID, awstypes.EdgeNodeSubnetRole)))
+		}
+	}
+
+	// Subnets that are in Local or Wavelength Zones must only have EdgeNode role.
+	for _, edgeSubnet := range subnetDataGroups.Edge {
+		for _, role := range edgeSubnet.Roles {
+			if role.Type != awstypes.EdgeNodeSubnetRole {
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(edgeSubnet.Idx), edgeSubnet.ID,
+					fmt.Sprintf("subnet %s must only be assigned role %s since it is in a Local or WaveLength Zone", edgeSubnet.ID, awstypes.EdgeNodeSubnetRole)))
+				break
+			}
+		}
+	}
+
+	return allErrs
+}
+
+// validateUntaggedSubnets ensures there are no additional untagged subnets in the BYO VPC.
+// An untagged subnet is a subnet without tag kubernetes.io/cluster/<cluster-id>.
+// Untagged subnets may be selected by the CCM, leading to various bugs, RFEs, and support cases.
```

Comment:
>leading to various bugs, RFEs, and support cases. 
what about to mention the bug/RFE/EP section related to this topic?

Reply:
Updated in the latest commit. I will squash those fixups when the PR is good to go :D